ecommerce/extensions/payment/utils.py

```
@@ -53,3 +66,73 @@ def clean_field_value(value):
+        details are logged as a new SDNCheckFailure object and the user's
+        account is deactivated. The check passes if:
+            * waffle flag for the SDN check is disabled.
+            * request to the SDN API times out
+            * SDN API returns a non-200 status code response
+            * user is not found on the SDN list
+
+        Args:
+            request (WSGIRequest): The request from which the site configuration is retrieved.
+            name (str): User's full name.
+            country (str): ISO alpha-2 country code where the user is from.
+        Returns:
+            The number of matches that have been found on the SDN list for the user.
+        """
+        if waffle.flag_is_active(request, SDN_CHECK_FLAG_NAME):
+            params = urlencode({
+                'sources': self.sdn_list,
+                'api_key': self.api_key,
+                'type': 'individual',
+                'name': name,
+                'country': country
+            })
+            sdn_check_url = '{api_url}?{params}'.format(
+                api_url=self.api_url,
+                params=params
+            )
+            user = request.user
+            try:
+                response = requests.get(sdn_check_url, timeout=settings.SDN_CHECK_REQUEST_TIMEOUT)
+            except requests.exceptions.Timeout:
+                logger.warning('Connection to US Treasury SDN API timed out for user [%d].', user.id)
+                return None
+
+            if response.status_code != 200:
+                logger.warning(
+                    'Unable to connect to US Treasury SDN API for user [%d]. Status code [%d] with message: [%s]',
+                    user.id, response.status_code, response.content
+                )
+                return None
+            elif json.loads(response.content)['total'] == 0:
+                return None
+
+            sdn_failure = SDNCheckFailure.objects.create(
```

Should the ``search`` method raise ``Timeout``? The idea was to silently fail and log the failure. That way if the Treasury API goes down it won't cause all payments to fail on our side.